src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Usage/ProvideCorrectArgumentToEnumHasFlag.cs

```diff
@@ -58,7 +58,9 @@ public override void Initialize(AnalysisContext context)
                         invocation.Arguments.Length != 1 ||
                         invocation.Instance == null ||
                         invocation.TargetMethod.Name != "HasFlag" ||
-                        invocation.Arguments[0].Value is not IConversionOperation conversion)
+                        invocation.Arguments[0].Value is not IConversionOperation conversion ||
+                        invocation.Instance.Type.TypeKind == TypeKind.TypeParameter ||
+                        conversion.Operand.Type.TypeKind == TypeKind.TypeParameter)
```

Comment:
Are we sure that `Instance.Type` and `Operand.Type` are both non-null? Might be safer to have null checks..